lib/Service/MailManager.php

```diff
@@ -321,17 +321,43 @@ public function moveMessage(Account $sourceAccount,
 	 */
 	public function deleteMessage(Account $account,
 		string $mailboxId,
-		int $messageId): void {
-		$this->eventDispatcher->dispatch(
-			BeforeMessageDeletedEvent::class,
-			new BeforeMessageDeletedEvent($account, $mailboxId, $messageId)
-		);
-
+		int $messageUid): void {
 		try {
 			$sourceMailbox = $this->mailboxMapper->find($account, $mailboxId);
 		} catch (DoesNotExistException $e) {
 			throw new ServiceException("Source mailbox $mailboxId does not exist", 0, $e);
 		}
+
+		$client = $this->imapClientFactory->getClient($account);
+		try {
+			$this->deleteMessageWithClient($account, $sourceMailbox, $messageUid, $client);
+		} finally {
+			$client->logout();
+		}
+	}
+
+	/**
+	 * @param Account $account
+	 * @param Mailbox $mailbox
+	 * @param int $messageId
```

Comment:
```suggestion
	 * @param int $messageUid
```

However, the parameters are typed and therefore the PHPDoc superfluous.

Reply:
Very true. Done.